src/m_pd.c

```diff
@@ -50,64 +64,99 @@ void gobj_save(t_gobj *x, t_binbuf *b)
 
 /* deal with several objects bound to the same symbol.  If more than one, we
 actually bind a collection object to the symbol, which forwards messages sent
-to the symbol. */
+to the symbol.
+CHR: We make a copy of the bindlist before messaging, so we can safely
+unbind symbols while iterating over the bindlist. We allocate on the stack
+up to a certain limit where we switch to heap allocation instead */
 
-static t_class *bindlist_class;
+static PERTHREAD int stackcount = 0;
```

Comment:
Shouldn't `stackcount` be PERINSTANCE rather than PERTHREAD?
(Of course PERINSTANCE doesn't exist unfortunately... I mean adding it to instance stuff)

Reply:
That's not necessary. Recursive Pd message calls always stay in the same thread. Since the stack count always goes back to 0, there is no state that has to be preserved. See also https://github.com/pure-data/pure-data/blob/ade0160ca291f7fb010ad15c973ed7fce7e1aae5/src/m_obj.c#L353

Reply:
Yes I also saw the `stackcount` in `m_obj.c`, and I had the same interrogation ;-)
Now I understand that it's not persistent data, so there's no risk two pd instances in the same thread could use the same data concurrently. Thanks  for the explanation!